### PR TITLE
Spark 3.2: #6041 follow-up/cleanup

### PR DESCRIPTION
**Pull request by wypoon**
In #6041, I ended up with a slightly different version of `TestSparkReaderDeletes` for v3.2 than in v3.3 because I created the PR before #6026 (the original PR, for v3.3) was merged and I then made some tweaks to `TestSparkReaderDeletes` for #6026. Here I align the version of `TestSparkReaderDeletes` in v3.2 with the version in v3.3. For future backports, it is easier when the 3.2 code base is the same as the 3.3 code base where there is no reason to be different.

### CODE REVIEW

**Comment by wypoon** [comment]
@flyrain sorry to trouble you again -- can you please review this? There were some simple improvements to `TestSparkReaderDeletes` in the v3.3 version that are missing from the v3.2 version. It's not a big deal, but would be nice to keep the code the same.

File: spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java [hunk 1]
```diff
@@ -533,55 +534,52 @@ public void testIsDeletedColumnWithoutDeleteFile() {
   public void testPosDeletesOnParquetFileWithMultipleRowGroups() throws IOException {
     Assume.assumeTrue(format.equals("parquet"));
 
-    String tblName = "test_multiple_row_groups";
+    String tblName = "test3";
```
Comment (flyrain) [inline spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java:537]: Kinda prefer meaningful name, like `table_with_multiple_row_groups`

**Review by flyrain — APPROVED** [review]
LGTM

File: spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java (hunk 1 above)
Comment (wypoon) [reply spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java:537]: The reason I changed it to `test3` in the original (3.3) PR was that in the base class, the tables were `test` and `test2`, so I used `test3` for this new table to be consistent!

**Comment by flyrain** [comment]
Merged. Thanks @wypoon!

**Pull request merged**